### PR TITLE
Download genotype

### PR DESCRIPTION
**Pull request by leehart**
Starts to address #509

### CODE REVIEW

File: webapp/src/js/components/panoptes/genome/tracks/GenotypesChannel.js [hunk 1]
```diff
@@ -480,6 +484,41 @@ const GenotypesControls = React.createClass({
     layoutGaps: React.PropTypes.bool,
   },
 
+  // FIXME: Ideally wanted to use <QueryString> to convert Query to String
```
Comment (benjeffery) [inline webapp/src/js/components/panoptes/genome/tracks/GenotypesChannel.js]: You can use the logic from <QueryString>, but you are best off extracting that out to a module (not a component) that both this and <QueryString> use.

File: webapp/src/js/components/panoptes/genome/tracks/GenotypesChannel.js [hunk 2]
```diff
@@ -480,6 +484,41 @@ const GenotypesControls = React.createClass({
...
+      colTableCapNamePlural: this.config.tablesById[this.config.twoDTablesById[this.props.table].columnDataTable].capNamePlural,
+      rowTableCapNamePlural: this.config.tablesById[this.config.twoDTablesById[this.props.table].rowDataTable].capNamePlural,
+      columnQueryAsString: this.convertQueryToString(this.config.tablesById[this.config.twoDTablesById[this.props.table].columnDataTable], this.props.columnQuery),
+      rowQueryAsString: this.convertQueryToString(this.config.tablesById[this.config.twoDTablesById[this.props.table].rowDataTable], this.props.rowQuery)
+    };
+
+    DataDownloader.downloadGenotypeData({...params, ...this.props});
+  },
+
+  handleDownloadLimitBreach(payload) {
```
Comment (benjeffery) [inline webapp/src/js/components/panoptes/genome/tracks/GenotypesChannel.js]: You don't need this - all the data is already on the client

File: webapp/src/js/components/containers/GenomeBrowserWithActions.js [hunk 3]
```diff
@@ -215,7 +215,7 @@ let GenomeBrowserWithActions = React.createClass({
                   title={sidebar ? 'Expand' : 'Sidebar'}
             />
           </div>
-          <GenomeBrowser setProps={setProps} sideWidth={150} {...subProps} />
+          <GenomeBrowser ref={(ref) => this.genomeBrowser = ref} setProps={setProps} sideWidth={150} {...subProps} />
```
Comment (leehart) [inline webapp/src/js/components/containers/GenomeBrowserWithActions.js]: No, sorry, that's an artefact from when I falsely assumed we'd be downloading all of the Genome Browser data.
I'll remove it.

File: webapp/src/js/components/panoptes/genome/tracks/GenotypesChannel.js (hunk 1 above)
Comment (leehart) [inline webapp/src/js/components/panoptes/genome/tracks/GenotypesChannel.js]: OK, I'll pursue that approach, thanks.
I ran into some hurdles on the first attempt, but expect we can overcome them; I'll keep you posted.

File: webapp/src/js/components/panoptes/genome/tracks/GenotypesChannel.js (hunk 2 above)
Comment (leehart) [inline webapp/src/js/components/panoptes/genome/tracks/GenotypesChannel.js]: ah yes, well caught 👍 
another relic from another earlier misapprehension. Will remove, thanks!

**Pull request closed without merging**